eval_converters/helm/adapter.py

```diff
@@ -124,385 +73,233 @@ def get_main_metric_name(run_path: str, schema_path: str) -> str:
-			if prompt_class == PromptClass.MultipleChoice: 
-				choices = [{"id": k, "text": v} for k, v in output_mapping_dict.items()]
-
-				classification_fields = {
-					"full_input": request_state.request.prompt,
-					"question": request_state.instance.input.text,
-					"choices": choices,
-					"ground_truth": ground_truth,
-				}
-			
-			instance = Instance(
-				task_type=TaskType.classification if prompt_class == PromptClass.MultipleChoice else TaskType.generation,
-				raw_input=request_state.instance.input.text,
-				language='en',  # FIXME: other languages?
-				sample_identifier=sample_identifier,
-				classification_fields=classification_fields,
-			)
-			
-			# 4. Output
-			output = Output(
-				response=request_state.result.completions[0].text
-			)
 
-			# 5. Evaluation
-			adapter = get_adapter_class_from_method_string(adapter_spec.method)
-			evaluation_method = EvaluationMethod(
-				method_name=adapter_spec.method,
-				description=adapter.__class__.__doc__, # Use the adapter's docstring as description
+		metric_names = []
+		for metric_spec in run_spec.metric_specs: 
+			metric_names.extend(
+				metric_spec.args.get('names') if metric_spec.args else []
 			)
 
-			try:
-				schema_path = os.path.join(base_dir, "schema_capabilities.yaml")
-				if not os.path.exists(schema_path):
-					wget.download("https://raw.githubusercontent.com/stanford-crfm/helm/main/src/helm/benchmark/static/schema_capabilities.yaml", schema_path)
-
-				schema = read_schema(schema_path)
-				main_metric_name = None
-
-				# find the main metric name from the schema
-				for group in run_spec.groups:
-					if group in schema.name_to_run_group and "main_name" in schema.name_to_run_group[group].environment:
-						main_metric_name = schema.name_to_run_group[group].environment["main_name"]
-						break
-				if main_metric_name is None:
-					raise Exception("Could not find main metric name")
-
-				# get the per instance stats from the data
-				per_instance_stats = []
-				if "per_instance_stats" in data:
-					per_instance_stats = from_json(data["per_instance_stats"], List[PerInstanceStats])
-
-				# get the instance scores from the per instance stats
-				instance_scores = {}
-				for per_instance_stat in per_instance_stats:
-					for stat in per_instance_stat.stats:	
-						if stat.name.name == main_metric_name:
-							assert stat.mean is not None
-							instance_scores[per_instance_stat.instance_id] = stat.mean
-							break
-
-			except Exception as e:
-				self.logger.warning(f"Error getting instance scores: {e}")
-				instance_scores = {}
-
-			score = instance_scores.get(request_state.instance.id, 0.0)
-
-			# 6. EvaluationResult
-			evaluation = Evaluation(
-				evaluation_method=evaluation_method,
-				ground_truth=ground_truth["text"],
-				score=score,
+		for metric_name in metric_names:
+			metric_config = MetricConfig(
+				evaluation_description=metric_name,
+				lower_is_better=False
 			)
+
+			for stat in stats:
```

Comment:
Might want to consider filtering down to a subset of relevant stats in the future.